test/cat.jl

```diff
@@ -88,27 +88,18 @@ module TestCat
     dt[1] = zeros(Int, nrow(dt))
     vcat(dt, alt_dt)
 
-    # Don't fail on non-matching names
-    names!(alt_dt, [:A, :B, :C])
-    vcat(dt, alt_dt)
-
     dtr = vcat(dt4, dt4)
     @test size(dtr, 1) == 8
     @test names(dt4) == names(dtr)
     @test isequal(dtr, [dt4; dt4])
 
-    dtr = vcat(dt2, dt3)
```

Comment:
This is now described in the docstrings https://github.com/JuliaData/DataTables.jl/pull/30/files#diff-4e49f4ea07be8309b10dab0820850f04R714 and the REPL help currently will look like this
```julia
help?> vcat
search: vcat hvcat VecOrMat DenseVecOrMat StridedVecOrMat AbstractVecOrMat levicivita @vectorize_2arg @vectorize_1arg

  vcat(A...)

  Concatenate along dimension 1.

  julia> a = [1 2 3 4 5]
  1×5 Array{Int64,2}:
   1  2  3  4  5

  julia> b = [6 7 8 9 10; 11 12 13 14 15]
  2×5 Array{Int64,2}:
    6   7   8   9  10
   11  12  13  14  15

  julia> vcat(a,b)
  3×5 Array{Int64,2}:
    1   2   3   4   5
    6   7   8   9  10
   11  12  13  14  15

  julia> c = ([1 2 3], [4 5 6])
  (
  [1 2 3],
  <BLANKLINE>
  [4 5 6])

  julia> vcat(c...)
  2×3 Array{Int64,2}:
   1  2  3
   4  5  6

  vcat(dts::AbstractDataTable...)

  Vertically concatenate AbstractDataTables with matching columns.

  julia> dt1 = DataTable(A=1:3, B=1:3); dt2 = DataTable(A=4:6, B=4:6); dt3 = DataTable(A=7:9, B=7:9, C=7:9);

  julia> vcat(dt1, dt2)
  6×2 DataTables.DataTable
  │ Row │ A │ B │
  ├─────┼───┼───┤
  │ 1   │ 1 │ 1 │
  │ 2   │ 2 │ 2 │
  │ 3   │ 3 │ 3 │
  │ 4   │ 4 │ 4 │
  │ 5   │ 5 │ 5 │
  │ 6   │ 6 │ 6 │

  julia> vcat(dt1, dt2, dt3)
  ERROR: ArgumentError: columns (A, B) of input(s) (1, 2) != columns (A, B, C) of input(s) (3)
```

example error for out of order columns
```julia
julia> vcat(DataTable(A = 1:3, B = 1:3), DataTable(B = 1:3, A = 1:3))
ERROR: ArgumentError: columns (A, B) of input(s) (1) != columns (B, A) of input(s) (2)
```